lab-data/yocto-stm32/bootlin-lab-data/nunchuk/linux/0003-input-joystick-wiichuck-Fix-stm32-building.patch

```diff
@@ -0,0 +1,46 @@
+From 6e0b53c13552db76b061c3cab132d228916f1761 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Myl=C3=A8ne=20Josserand?= <mylene.josserand@bootlin.com>
+Date: Thu, 17 Oct 2019 17:17:03 +0200
+Subject: [PATCH] input: joystick: wiichuck: Fix stm32 building
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+With linux-4.19.49 for stm32 plateform, the build of this
+kernel is failing with the following error:
+
+WARNING: vmlinux.o(.data+0x7a6bc): Section mismatch in reference from the variable wiichuck_driver to the function .init.text:wiichuck_probe()
+The variable wiichuck_driver references
+the function __init wiichuck_probe()
+If the reference is valid then annotate the
+variable with __init* or __refdata (see linux/init.h) or name the variable:
+*_template, *_timer, *_sht, *_ops, *_probe, *_probe_one, *_console
+
+FATAL: modpost: Section mismatches detected.
+Set CONFIG_SECTION_MISMATCH_WARN_ONLY=y to allow them.
+
+Removing the __init reference fixes it.
+
+Signed-off-by: Mylène Josserand <mylene.josserand@bootlin.com>
+---
+ drivers/input/joystick/wiichuck.c | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/drivers/input/joystick/wiichuck.c b/drivers/input/joystick/wiichuck.c
+index 304738eb9..ab17fd7f5 100644
+--- a/drivers/input/joystick/wiichuck.c
++++ b/drivers/input/joystick/wiichuck.c
+@@ -280,8 +280,8 @@ static void wiichuck_open(struct input_polled_dev *poll_dev)
+ 	wiichuck->poll_dev->poll_interval = WIICHUCK_POLL_PERIOD;
+ }
+ 
+-static int __init wiichuck_probe(struct i2c_client *client,
+-				const struct i2c_device_id *id)
++static int wiichuck_probe(struct i2c_client *client,
++			  const struct i2c_device_id *id)
+ {
+ 	struct wiichuck_device *wiichuck;
+ 	struct input_polled_dev *poll_dev;
+-- 
+2.11.0
+
```

Comment:
The changes is correct and should be merged in patch 0001

Reply:
Sure, I will update it, thanks.